### DISCUSSION TITLE
Bottom sheet with list, fixed header & input

### DISCUSSION THREAD

**Discussion by filippobarcellos** [mateusz1913/react-native-avoid-softinput#75] [Q&A] [answered]
### Feature request

Hi. First of all thank you for the lib, it's really awesome. I'm having some problems trying to make the bottom sheet to respond based on the size of the content. My bottom sheet needs to have a fixed input on the bottom and 2 positions - 50% and 90%. On 50% the bottom sheet needs to be pushed when the keyboard appears and on 90% the content of my flat list needs to be pushed up and scrolled to the last item visible. Is possible to achieve this? 

 ![Untitled-2022-04-23-0039](https://user-images.githubusercontent.com/56128203/164841541-7f5e4b08-7d29-4ef5-91f9-f24f9c63d1e9.png)

## Steps To Achieve

1. On 50% - bottom sheet needs to be pushed up until the max size(90%)
2. on 90% - the flat list inside needs to be pushed up

**Comment by mateusz1913** [comment, marked as answer]
Hi @filippobarcellos, I moved it to the discussions section because there is nothing to be done from the library perspective. However use case is very interesting, so I'll try to recommend sth here.

To simplify, I'll refer to 50% & 90% positions as "half-screen" bottom sheet and "fullscreen" bottom sheet.

I have 2 recommended ways instead:
1) With `AvoidSoftInput` module **_disabled_** (disabled, because you will be handling it manually), you can use [`useSoftInputHeightChanged`](https://mateusz1913.github.io/react-native-avoid-softinput/docs/api/hooks/use-soft-input-height-changed) for both iOS & Android, to detect current keyboard height and apply that height as an additional bottom padding to bottom sheet's content view. (⚠️ to make it work on Android, you have to call `AvoidSoftInput.setAdjustNothing` )
2) For iOS use 1) and for Android, just make sure that your android activity's [`windowSoftInputMode`](https://developer.android.com/guide/topics/manifest/activity-element#wsoft) attribute is set to `adjustResize` and `AvoidSoftInput` module is **_disabled_** - that way, you let Android OS handle resizing activity window & keyboard will never obscure your bottom sheet layout.

You may ask, why you have to handle it manually.

Your use case is quite complex and if using `AvoidSoftInput` module or `AvoidSoftInputView` component, it would have to be handled differently depending on if it's a half-screen or fullscreen bottom sheet. Also, I don't know, which solution you are using for creating/displaying your bottom sheet, which makes it even harder to predict if the final result will be satisfying for you.
In such complex scenarios, you will want to have full control over the way additional offset is applied to your bottom sheet when the keyboard is displayed.
Additionally, you mentioned the requirements for FlatList - 1st with pushing up, it will have to be handled with proper flex styling, 2nd with scrolling to the end, you have to do it manually anyway (so nothing to be done from lib's POV).

I hope this will help, if you have any more questions, feel free to ask here or open a new discussion in [Discussions tab](https://github.com/mateusz1913/react-native-avoid-softinput/discussions)
